File: storage/data_original/at_coder/agc006/B/4415089.py
# coding:utf-8

import sys

INF = float('inf')
MOD = 10 ** 9 + 7

def LI(): return [int(x) for x in sys.stdin.readline().split()]
def LI_(): return [int(x) - 1 for x in sys.stdin.readline().split()]
def LS(): return sys.stdin.readline().split()
def II(): return int(sys.stdin.readline())
def SI(): return input()


def main():
    n, x = LI()

    if x == 1 or x == 2 * n - 1:
        print('No')
        return 0

    print('Yes')
    if n < 3:
        print(*[n for n in range(1, 2 * n)], sep='\n')
        return 1

    # n??????[x-1, x, x+1]???
    tmp = [x - 1, x, x + 1]

    remain = list(set([n for n in range(1, 2 * n)]) - set(tmp))
    k = n - 2
    tmp = remain[:k] + tmp + remain[k:]
    print(*tmp, sep='\n')

    return 1


main()

File: data/data_transpiled/at_coder/agc006/B/4415089.java
static final double INF = Double . MAX_VALUE ;
final double MOD = 10 * * 9 + 7 ;
{
  int [ ] LI = new int [ 3 ] ;
  int [ ] LI_ = new int [ 3 ] ;
  int [ ] LS = new int [ 3 ] ;
  int [ ] II = new int [ 3 ] ;
  int [ ] SI = new int [ 3 ] ;
  {
    int n = LI [ 0 ] ;
    int x = LI_ [ 1 ] ;
    if ( x == 1 || x == 2 * n - 1 ) {
      System . out . println ( "No" ) ;
      return 0 ;
    }
    System . out . println ( "Yes" ) ;
  }
  if ( n < 3 ) {
    System . out . println ( Arrays . toString ( LENGTH ) ) ;
    return 1 ;
  }
  else {
    return 0 ;
  }
}
